IOPool/Input/src/RootInputFileSequence.cc

```diff
@@ -252,14 +252,15 @@ namespace edm {
       }
       for (std::vector<std::string>::const_iterator it = fNames.begin(); it != fNames.end(); ++it) {
         try {
+          usedFallback_ = (it != fNames.begin());
```

Comment:
The bug fix is entirely in this file. The changes in other files are for the unit test and related to converting the seg fault to an exception.